compositor_pipeline/src/pipeline/whip_whep/whip_handlers/terminate_whip_session.rs

```diff
@@ -0,0 +1,63 @@
+use crate::pipeline::whip_whep::{
+    error::WhipServerError, validate_bearer_token::validate_token, WhipWhepState,
+};
+use axum::{
+    extract::{Path, State},
+    http::{HeaderMap, StatusCode},
+};
+use compositor_render::InputId;
+use std::sync::Arc;
+use tracing::info;
+
+pub async fn handle_terminate_whip_session(
+    Path(id): Path<String>,
+    State(state): State<Arc<WhipWhepState>>,
+    headers: HeaderMap,
+) -> Result<StatusCode, WhipServerError> {
+    let bearer_token: Option<String>;
+    let input_id = InputId(Arc::from(id));
+
+    if let Ok(connections) = state.input_connections.lock() {
+        if let Some(connection) = connections.get(&input_id) {
+            bearer_token = connection.bearer_token.clone();
+        } else {
+            return Err(WhipServerError::NotFound(format!(
+                "InputID {input_id:?} not found"
+            )));
+        }
+    } else {
+        return Err(WhipServerError::InternalError(
+            "Input connections lock error".to_string(),
+        ));
+    }
+    validate_token(bearer_token, headers.get("Authorization")).await?;
+    let peer_connection;
+
+    if let Ok(mut connections) = state.input_connections.lock() {
+        if let Some(connection) = connections.get_mut(&input_id) {
+            peer_connection = connection.peer_connection.clone();
+            connection.peer_connection = None;
+            drop(connection.audio_sender.clone());
```

Comment:
no idea what you wanted achive here, but it does not make sense. You are cloning value and then droping that cloned value.